### PR TITLE
port sklearn feature_selection

### PR DESCRIPTION
**Pull request by cxhernandez**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

[Describe changes here]

re: https://github.com/msmbuilder/msmbuilder/issues/797

Expanding msmbuilder pipeline functionality by porting feature selection algorithms from sklearn.

### CODE REVIEW

**Comment by cxhernandez** [comment]
I'm noticing that [quite a few of these classes](http://scikit-learn.org/stable/modules/classes.html#module-sklearn.feature_selection) require some sort of response variable for either classification or regression, which doesn't really seem applicable to our problem scope..... I'll try to exclude these (e.g. `chi2`)

Some of them also require a scoring function, which I can only imagine in this case could be `tICA.score` or `PCA.score`?

**Comment by mpharrigan** [comment]
Yeah let's be careful about only exposing things people might actually find useful!

**Comment by cxhernandez** [comment]
@mpharrigan: Would it be okay to remove `msmbuilder.featurizer.FeatureUnion`?

**Comment by mpharrigan** [comment]
Sounds good to me

**Comment by cxhernandez** [comment]
Also, some of these, like `FeatureSelector` and `FeatureUnion`, will take lists of trajectories while the rest can only handle lists of `ndarray` objects.... Not sure if maybe we want to think about moving `FeatureSelector` and `FeatureUnion` to `msmbuilder.featurizer` (which perhaps we should rename to `feature_extraction` to be [consistent with scipy](http://scikit-learn.org/stable/modules/feature_extraction.html)?)

**Comment by cxhernandez** [comment]
okay, I think this is pretty much ready for review

File: docs/changelog.rst [hunk 1]
```diff
@@ -11,10 +11,21 @@ This is the current development version of MSMBuilder
 API Changes
 ~~~~~~~~~~~
 
+- ``msmbuilder.featurizer.FeatureUnion`` is now deprecated. Please used
```
Comment (mpharrigan) [inline docs/changelog.rst]: used -> use

File: msmbuilder/feature_selection/base.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
+# Author: Carlos Xavier Hernandez <cxh@stanford.edu>
```
Comment (msultan) [inline msmbuilder/feature_selection/base.py]: is this code significantly different than the cluster version? Why not just inherent from that directly instead of copying it over.

Comment (cxhernandez) [inline msmbuilder/feature_selection/base.py]: The docstrings should be different at least.

Comment (msultan) [inline msmbuilder/feature_selection/base.py]: maybe we should have a timeseries to array concatenator and everything inherits from it. The descriptor can be different but then use the super method to initiate the base class.

Comment (cxhernandez) [inline msmbuilder/feature_selection/base.py]: Is this better?

Comment (msultan) [inline msmbuilder/feature_selection/base.py]: we can move the clusterer thing to a super class later on.

File: msmbuilder/feature_selection/featureselector.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Author: Carlos Xavier Hernandez <cxh@stanford.edu>
+# Contributors:
+# Copyright (c) 2016, Stanford University and the Authors
+# All rights reserved.
+import numpy as np
+
+from .base import MultiSequenceFeatureSelectionMixin
+
+
+class FeatureSelector(MultiSequenceFeatureSelectionMixin):
```
Comment (msultan) [inline msmbuilder/feature_selection/featureselector.py]: We should add a warning that you need to normalize the dataset afterwards. Mixing and matching featurizers with differing means and variances is gonna cause a lot of issues later on. Alternate option is to add a sklearn normalizer as a attribute to the object and it get auto called when fitting.

**Comment by msultan** [comment]
thanks for doing this @cxhernandez

File: msmbuilder/feature_selection/featureselector.py (hunk 3 above)
Comment (cxhernandez) [inline msmbuilder/feature_selection/featureselector.py]: I'll make a note in the docstring, but I think a warning might be a tad much. @mpharrigan thoughts?

File: msmbuilder/feature_selection/featureselector.py [hunk 4]
```diff
@@ -14,6 +14,9 @@ class FeatureSelector(MultiSequenceFeatureSelectionMixin):
     concatenates the results. This is useful to combine several feature
     extraction mechanisms into a single transformer.
 
+    Note: Users should consider using `msmbuilder.preprocessing.StandardScaler`
```
Comment (msultan) [inline msmbuilder/feature_selection/featureselector.py]: maybe i am being paranoid but a stronger warning is probably warranted. 
```
WARNING: Users are highly encouraged to use msmbuilder.preprocessing.StandardScaler to normalize the outputs ```

**Comment by mpharrigan** [comment]
lgtm

**Comment by cxhernandez** [comment]
@mpharrigan: can this one be merged as well?
